projects/components/src/navigation/navigation-list-component.service.ts

```diff
@@ -29,11 +34,10 @@ export class NavigationListComponentService {
     return updatedItems;
   }
 
-  public resolveNavItemConfigTimeRanges(
-    navItems: NavItemConfig[],
-    pageLevelTimeRangeFeatureState: FeatureState
-  ): Observable<NavItemConfig[]> {
-    return combineLatest(this.getTimeRangesForNavItems(navItems, pageLevelTimeRangeFeatureState));
+  public resolveNavItemConfigTimeRanges(navItems: NavItemConfig[]): Observable<NavItemConfig[]> {
+    return this.featureStateResolver
+      .getFeatureState(ApplicationFeature.PageTimeRange)
+      .pipe(switchMap(featureState => combineLatest(this.getTimeRangesForNavItems(navItems, featureState))));
```

Comment:
So any time a time range is saved, you would get a whole new set of nav items?

Reply:
Yes, is there a better operator for this ?

Reply:
This gets back to the suggestion to do the resolution on click rather than up front, but we can revisit later as discussed. Just need to make sure with this approach that a new set of items doesn't cause issues (e.g. losing the active item, flickering etc.)